protos/telemetry/telemetry.proto

```diff
@@ -286,3 +295,47 @@ enum LandedState{
     LANDED_STATE_TAKING_OFF = 3;
     LANDED_STATE_LANDING = 4;
 }
+
+// Odometry message type.
+message Odometry {
+    // Mavlink frame id
+    enum MavFrame {
+        UNDEF = 0;
+        MAV_FRAME_BODY_NED = 8; // Setpoint in body NED frame. This makes sense if all position control is externalized - e.g. useful to command 2 m/s^2 acceleration to the right.
+        MAV_FRAME_VISION_NED = 16; // Odometry local coordinate frame of data given by a vision estimation system, Z-down (x: north, y: east, z: down).
+        MAV_FRAME_ESTIM_NED = 18; // Odometry local coordinate frame of data given by an estimator running onboard the vehicle, Z-down (x: north, y: east, z: down).
+    }
+
+    uint64 time_usec = 1; // Timestamp (0 to use Backend timestamp).
+    MavFrame frame_id = 2; // Coordinate frame of reference for the pose data.
+    MavFrame child_frame_id = 3;  // Coordinate frame of reference for the velocity in free space (twist) data.
+    PositionXyz position = 4; // Position.
+    Quaternion q = 5; // Quaternion components, w, x, y, z (1 0 0 0 is the null-rotation).
+    SpeedBody speed_body = 6; // Linear speed (m/s).
+    AngularVelocityBody angular_velocity_body = 7; // Angular speed (rad/s).
+    Covariance pose_covariance = 8; // Pose cross-covariance matrix.
+    Covariance velocity_covariance = 9; // Velocity cross-covariance matrix.
+}
+
+/* Covariance type.
+ * Row-major representation of a 6x6 cross-covariance matrix
+ * upper right triangle.
+ * Set first to NaN if unknown.
+ */
+message Covariance {
+    repeated float covariance_matrix = 1;
+}
+
+// Speed type, represented in the Body (X Y Z) frame and in metres/second.
+message SpeedBody {
+    float velocity_x_m_s = 1; // Velocity in X in metres/second
+    float velocity_y_m_s = 2; // Velocity in Y in metres/second
+    float velocity_z_m_s = 3; // Velocity in Z in metres/second
+}
+
+// Position type, represented in the Body (X Y Z) frame
+message PositionXyz {
```

Comment:
@JonasVautherin what about `PositionBody`? Like `SpeedBody` and `AngularVelocityBody`

Reply:
I vote `PositionBody`.